main.rb

```diff
@@ -0,0 +1,107 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    array = to_a
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << i if yield i
+    end
+    arr
+  end
+
+  def my_all?(*args, &block)
+    if block_given?
+      my_select(&block).length == length
+    else
+      return true if args == []
+
+      return (my_select { |i| i.instance_of?(args[0]) }).length == length if args[0].is_a? Class
+      return (my_select { |i| i.match(args[0]) }).length == length if args[0].is_a? Regexp
+      return (my_select { |i| i == args[0] }).length == length if args[0].is_a? Integer
+
+      false
+    end
+  end
+
+  def my_any?(*args, &block)
+    if block_given?
+      my_select(&block).length.positive?
+    else
+      return (my_select { |i| ![nil, false].include?(i) }).length.positive? if args == []
+      return (my_select { |i| i == args[0] }).length.positive? if args[0].is_a? Integer
+      return (my_select { |i| i.instance_of?(args[0]) }).length.positive? if args[0].is_a? Class
+      return (my_select { |i| i.match(args[0]) }).length.positive? if args[0].is_a? Regexp
+    end
+  end
+
+  def my_none?(*args, &block)
+    if block_given?
+      my_select(&block).length.zero?
+    else
+      return length.zero? || (my_select { |i| ![nil, false].include?(i) }).length.zero? if args == []
+      return (my_select { |i| i.instance_of?(args[0]) }).length.zero? if args[0].is_a? Class
+      return (my_select { |i| i == args[0] }).length.zero? if args[0].is_a? Integer
+    end
+  end
```

Comment:
- [ ] when a Regex is passed as an argument it should return true if none of the collection matches the Regex

```Ruby
  words.my_none?(/z/)

  # expected true
  # got nil
```

- [ ] when a Regex is passed as an argument it should return false if any of the collection matches the Regex

```Ruby
  words.my_none?(/d/)

  # expected false
  # got nil
```